src/components/header/search-form/SearchForm.tsx

```diff
@@ -0,0 +1,47 @@
+import './SearchForm.css';
+import {searchFunction} from "../../../models/search";
+import {useState} from "react";
+
+function SearchForm({ searchChange }: {searchChange: searchFunction}) {
+    const [formData, setFormData] = useState({
+        query: '',
+    });
+    const onSearch = (event: any): void => {
+        searchChange(formData.query);
+        event.preventDefault();
+
+        console.log('SearchForm, onSearch', formData.query);
+    }
+
+    const onKeyDown = (event: any): void => {
+        if (event.key === 'Enter') {
+            searchChange(formData.query);
+        }
+    }
+
+    const onQueryChange = (event: any): void => {
+        const {name, value} = event.target;
+
+        setFormData((prevState) => ({...prevState, [name]: value}));
+
+        console.log('SearchForm, onQueryChange', event.target.value);
+    }
+
+    return (
+      <form className="wrap-block" onSubmit={onSearch}>
+          <div className="label">FIND YOUR MOVIE</div>
+          <input type="text"
+                 name="query"
+                 value={formData.query}
+                 onChange={onQueryChange}
+                 onKeyDown={onKeyDown}
+                 onFocus={onQueryChange}
```

Comment:
I don't need to change the state on input focus.

Reply:
Removed, thank you